server/portal/apps/webhooks/views.py

```diff
@@ -108,90 +122,20 @@ def post(self, request, *args, **kwargs):
                 client = user.tapis_oauth.client
                 job_details = client.jobs.getJob(jobUuid=job_id)
 
-                event_data[Notification.EXTRA]['remoteSubmitted'] = str(job['remoteSubmitted'])
-                event_data[Notification.EXTRA]['ended'] = str(job['remoteEnded'])
-                event_data[Notification.EXTRA]['archiveSystemId'] = job_details.archiveSystemId
-                event_data[Notification.EXTRA]['archiveSystemDir'] = job_details.archiveSystemDir
+                event_data[Notification.EXTRA]['remoteOutcome'] = job_details.remoteOutcome
 
-            if job_status not in settings.PORTAL_JOB_NOTIFICATION_STATES:
-                logger.debug(
-                    "Job ID {} for owner {} entered {} state (no notification sent)".format(
-                        job_id, username, job_status
-                    )
-                )
-                return HttpResponse("OK")
+                try:
+                    logger.info('Indexing job output for job={}'.format(job_id))
+
+                    agave_indexer.apply_async(args=[job_details.archiveSystemId],
+                                              kwargs={'filePath': job_details.archiveSystemDir})
+                except Exception as e:
+                    logger.exception('Error indexing job output: {}'.format(e))
+                    return HttpResponse(json.dumps(e), content_type='application/json', status=400)
 
-            if job_status == 'FAILED':
```

Comment:
Turns out a lot of this was redundant. The only important difference of state is when FINISHED, the notification will push up the `remoteOutcome` to the frontend for parsing the output path

Reply:
Good point. It may be better to also key off of whether or not `job_details.remoteOutcome` is not `FAILED_SKIP_ARCHIVE`, and also check that the state is not `CANCELLED`